examples/webhook.php

```diff
@@ -8,67 +8,165 @@
 // Import Invoice client class.
 use BTCPayServer\Client\Invoice;
 
-// Fill in with your BTCPay Server data.
-$apiKey = '';
-$host = ''; // e.g. https://your.btcpay-server.tld
-$storeId = '';
-$secret = ""; // webhook secret configured in the BTCPay UI
-
-$myfile = fopen("BTCPay.log", 'ab');
-
-$raw_post_data = file_get_contents('php://input');
-
-$date = date('m/d/Y h:i:s a');
-
-if (false === $raw_post_data) {
-    fwrite($myfile, $date . " : Error. Could not read from the php://input stream or invalid BTCPayServer payload received.\n");
-    fclose($myfile);
-    throw new \Exception('Could not read from the php://input stream or invalid BTCPayServer payload received.');
-}
-
-$payload = json_decode($raw_post_data, false, 512, JSON_THROW_ON_ERROR);
-
-if (true === empty($payload)) {
-    fwrite($myfile, $date . " : Error. Could not decode the JSON payload from BTCPay.\n");
-    fclose($myfile);
-    throw new \Exception('Could not decode the JSON payload from BTCPay.');
-}
-
-// verify hmac256
-$headers = getallheaders();
-$sig = $headers['Btcpay-Sig'];
-
-if ($sig !== "sha256=" . hash_hmac('sha256', $raw_post_data, $secret)) {
-    fwrite($myfile, $date . " : Error. Invalid Signature detected! \n was: " . $sig . " should be: " . hash_hmac('sha256', $raw_post_data, $secret) . "\n");
-    fclose($myfile);
-    throw new \Exception('Invalid BTCPayServer payment notification message received - signature did not match.');
+class WebhookExample
+{
+    public $apiKey;
+    public $host;
+    public $storeId;
+    public $secret;
+    public $webhookId;
+
+    public function __construct()
+    {
+        // Fill in with your BTCPay Server data.
+        $this->apiKey = '';
+        $this->host = ''; // e.g. https://your.btcpay-server.tld
+        $this->storeId = '';
+        $this->secret = ''; // webhook secret as shown in the webhook UI / returned by createWebhook()
+        $this->webhookId = ''; // only needed for the updateWebhook() example.
+    }
+
+    public function processWebhook()
+    {
+        $myfile = fopen("BTCPay.log", 'ab');
+        $raw_post_data = file_get_contents('php://input');
+
+        $date = date('m/d/Y h:i:s a');
+
+        if (false === $raw_post_data) {
+            fwrite(
+                $myfile,
+                $date . " : Error. Could not read from the php://input stream or invalid BTCPayServer payload received.\n"
+            );
+            fclose($myfile);
+            throw new \Exception(
```

Comment:
Should always return \RuntimeException as a best practise. Generic \Exception is not to be used. Not sure why exactly.

Reply:
Interesting never read it anywhere, any source to this?

Reply:
fixed